go/vt/vtgate/planbuilder/testdata/union_cases.json

```diff
@@ -1083,35 +1098,41 @@
         ],
         "Inputs": [
           {
-            "OperatorType": "Distinct",
-            "Collations": [
-              "(0:1)"
-            ],
+            "OperatorType": "Projection",
+            "Expressions": null,
             "Inputs": [
               {
-                "OperatorType": "Concatenate",
+                "OperatorType": "Distinct",
```

Comment:
Looks like this Projection is not needed.
Compact can mostly merge the two projection operators here.